patches/expo-secure-store/expo-secure-store+14.2.4+001+additional-config-options.patch

```diff
@@ -0,0 +1,1114 @@
+       SecItemDelete(noAuthSearchDictionary as CFDictionary)
+     }
+ 
+-    Function("canUseBiometricAuthentication") {() -> Bool in
+-      #if os(tvOS)
+-      return false
+-      #else
+-      let context = LAContext()
+-      var error: NSError?
+-      let isBiometricsSupported: Bool = context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &error)
++    Function("canUseBiometricAuthentication") { () -> Bool in
++      return areBiometricsEnabled()
++    }
++  }
++
++  private func askForAuth(prompt: String) async throws {
++    let context = LAContext()
++    context.localizedReason = prompt
++    var error: NSError?
++    let isAuthenticationSupported: Bool = context.canEvaluatePolicy(
++      LAPolicy.deviceOwnerAuthentication,
++      error: &error
++    )
++    if error != nil {
++      throw error ?? RuntimeError("No authentication method available")
++    }
++    if !isAuthenticationSupported {
++      throw RuntimeError("No authentication method available")
++    }
++    let result = try await context.evaluatePolicy(
++      LAPolicy.deviceOwnerAuthentication,
++      localizedReason: context.localizedReason
++    )
+ 
+-      if error != nil {
+-        return false
++    if !result {
++      throw RuntimeError("Unable to authenticate")
++    }
++  }
++
++  private func getAuthType() -> AuthType {
++    if !areBiometricsEnabled() {
++      return AuthType.credentials
++    }
++
++    let biometryType = LAContext().biometryType
++
++    if biometryType == LABiometryType.none {
++      return AuthType.credentials
++    }
++
++    if biometryType == LABiometryType.faceID {
++      return AuthType.faceID
++    }
++
++    if biometryType == LABiometryType.touchID {
++      return AuthType.touchID
++    }
++
++    if #available(iOS 17.0, *) {
++      if biometryType == LABiometryType.opticID {
++        return AuthType.opticID
+       }
+-      return isBiometricsSupported
+-      #endif
+     }
++
++    return AuthType.credentials
++  }
++
++  private func getAccessControl(accessibility: CFString) throws -> SecAccessControl
++  {
++    guard
++      Bundle.main.infoDictionary?["NSFaceIDUsageDescription"] as? String
++        != nil
++    else {
++      throw MissingPlistKeyException()
++    }
++
++    var error: Unmanaged<CFError>? = nil
++    guard
++      let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, .userPresence, &error)
++    else {
++      let errorCode = error.map { CFErrorGetCode($0.takeRetainedValue()) }
++      throw SecAccessControlError(errorCode)
++    }
++
++    return accessOptions
++  }
++
++  private func areBiometricsEnabled() -> Bool {
++    #if os(tvOS)
++    return false
++    #else
++    let context = LAContext()
++    var error: NSError?
++    let isBiometricsSupported: Bool = context.canEvaluatePolicy(LAPolicy.deviceOwnerAuthenticationWithBiometrics, error: &error)
++
++    if error != nil {
++      return false
++    }
```

Comment:
Same as in one of the comments above. If there is an error `canEvaluatePolicy` should return `false` anyways